Submitted By:            Douglas R. Reno <renodr at linuxfromscratch dot org>
Date:                    2020-11-11
Initial Package Version: 3.34.0
Origin:                  Upstream PR !60 (filed by Xi Ruoyao)
Upstream Status:         Applied
Description:             Ports gnome-tweaks to use libhandy-1.x instead of
                         libhandy-0.x. See the changes in 
                         https://gitlab.gnome.org/GNOME/gnome-tweaks/-/merge_requests/60/diffs

diff -Naurp gnome-tweaks-3.34.0/gnome-tweaks gnome-tweaks-3.34.0/gnome-tweaks
--- gnome-tweaks-3.34.0/gnome-tweaks	2019-09-27 06:15:56.000000000 -0500
+++ gnome-tweaks-3.34.0/gnome-tweaks	2020-11-11 18:30:29.526638157 -0600
@@ -12,7 +12,7 @@ import sys
 
 import gi
 gi.require_version("Gtk", "3.0")
-gi.require_version("Handy", "0.0")
+gi.require_version("Handy", "1")
 
 import gtweak
 from gtweak.defs import VERSION
diff -Naurp gnome-tweaks-3.34.0/gtweak/tweakview.py gnome-tweaks-3.34.0/gtweak/tweakview.py
--- gnome-tweaks-3.34.0/gtweak/tweakview.py	2019-09-27 06:15:56.000000000 -0500
+++ gnome-tweaks-3.34.0/gtweak/tweakview.py	2020-11-11 18:32:15.805636986 -0600
@@ -23,8 +23,7 @@ class Window(Gtk.ApplicationWindow):
         self.hsize_group = Gtk.SizeGroup(mode=Gtk.SizeGroupMode.HORIZONTAL)
 
         self.main_box = Handy.Leaflet()
-        self.main_box.set_mode_transition_type(Handy.LeafletModeTransitionType.SLIDE)
-        self.main_box.set_child_transition_type(Handy.LeafletChildTransitionType.SLIDE)
+        self.main_box.set_transition_type(Handy.LeafletTransitionType.SLIDE)
 
         left_box = self.sidebar()
         right_box = self.main_content()
@@ -68,14 +67,13 @@ class Window(Gtk.ApplicationWindow):
     def titlebar(self):
 
         header = Handy.Leaflet()
-        header.set_mode_transition_type(Handy.LeafletModeTransitionType.SLIDE)
-        header.set_child_transition_type(Handy.LeafletChildTransitionType.SLIDE)
+        header.set_transition_type(Handy.LeafletTransitionType.SLIDE)
         header.connect("notify::visible-child", self._update_decorations)
         header.connect("notify::fold", self._update_decorations)
 
-        left_header = Gtk.HeaderBar()
+        left_header = Handy.HeaderBar()
         left_header.props.show_close_button = True
-        right_header = Gtk.HeaderBar()
+        right_header = Handy.HeaderBar()
         right_header.props.show_close_button = True
         right_header.props.hexpand = True
 
@@ -226,9 +224,9 @@ class Window(Gtk.ApplicationWindow):
     def _update_decorations(self, *_):
         header = self.get_titlebar()
         if header.props.folded:
-            self.header_group.set_focus(header.get_visible_child())
+            self.header_group.set_decorate_all(True)
         else:
-            self.header_group.set_focus(None)
+            self.header_group.set_decorate_all(False)
 
     def _after_key_press(self, widget, event):
         if not self.button.get_active() or not self.entry.is_focus():
diff -Naurp gnome-tweaks-3.34.0/README.md gnome-tweaks-3.34.0/README.md
--- gnome-tweaks-3.34.0/README.md	2019-09-27 06:15:56.000000000 -0500
+++ gnome-tweaks-3.34.0/README.md	2020-11-11 18:30:05.422638423 -0600
@@ -21,7 +21,7 @@ RUNTIME DEPENDENCIES
   - GLib (>= 2.58)
   - GTK+ 3 (>= 3.12)
   - gnome-desktop (>= 3.30)
-  - libhandy
+  - libhandy (>= 1.0)
   - libsoup
   - libnotify
   - Pango
